### PR TITLE
Jar-in-jar Fabric API deps

### PR DESCRIPTION
**Pull request by wizjany**
No longer need separate install of fabric api mod.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,7 +14,6 @@ Installation
 ------------
 
 1. Install [Fabric](https://fabricmc.net/use)
-2. Install [Fabric API mod](https://www.curseforge.com/minecraft/mc-mods/fabric-api) into Minecraft mods folder
 3. Install [this mod](https://www.curseforge.com/minecraft/mc-mods/worldeditcui-fabric) into Minecraft mods folder
```
Comment (wizjany) [inline README.md:17]: I know how to count. Really.

File: gradlew [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 #
-# Copyright � 2015-2021 the original authors.
+# Copyright © 2015-2021 the original authors.
```
Comment (octylFractal) [inline gradlew:4]: What's the actual change here? Is this good?

**Review by octylFractal — CHANGES_REQUESTED** [review]

File: gradlew (hunk 2 above)
Comment (wizjany) [reply gradlew:4]: changed the file encoding from cp1252 to utf-8

File: build.gradle.kts [hunk 3]
```diff
@@ -50,6 +50,8 @@ tasks.withType(JavaCompile::class) {
     options.compilerArgs.addAll(listOf("-Xlint:all", "-Xlint:-processing"))
 }
 
+val fabricApiConfiguration: Configuration = configurations.create("fabricApi")
```
Comment (zml2008) [inline build.gradle.kts:53]:
```suggestion
val fabricApi: Configuration by configurations.creating
```

File: build.gradle.kts [hunk 4]
```diff
@@ -59,12 +61,59 @@ dependencies {
...
     })
     modImplementation("net.fabricmc:fabric-loader:$fabricLoaderVersion")
-    modImplementation("net.fabricmc.fabric-api:fabric-api:$fabricApiVersion")
     modImplementation("com.terraformersmc:modmenu:$modmenuVersion")
     modImplementation("net.earthcomputer.multiconnect:multiconnect-api:$multiconnectVersion") {
         isTransitive = false
     }
 
+    // [1] declare fabric-api dependency...
+    "fabricApi"("net.fabricmc.fabric-api:fabric-api:$fabricApiVersion")
```
Comment (zml2008) [inline build.gradle.kts:70]:
```suggestion
    fabricApi("net.fabricmc.fabric-api:fabric-api:$fabricApiVersion")
```

File: build.gradle.kts [hunk 5]
```diff
@@ -59,12 +61,59 @@ dependencies {
...
+    val wantedDependencies = (fabricModJson["depends"] ?: error("no depends in fabric.mod.json")).keys
+        .filter { it == "fabric-api-base" || it.contains(Regex("v\\d$")) }
+        .map { "net.fabricmc.fabric-api:$it" }
+        .toSet()
+    logger.lifecycle("Looking for these dependencies:")
+    for (wantedDependency in wantedDependencies) {
+        logger.lifecycle(wantedDependency)
+    }
+    // [3] and now we resolve it to pick out what we want :D
+    val fabricApiDependencies = fabricApiConfiguration.incoming.resolutionResult.allDependencies
```
Comment (zml2008) [inline build.gradle.kts:86]:
```suggestion
    val fabricApiDependencies = fabricApi.incoming.resolutionResult.allDependencies
```

**Review by zml2008 — CHANGES_REQUESTED** [review]
some minor naming tweaks here, just going off memory

**Review by zml2008 — APPROVED** [review]
good enough for now, WE's been using the same thing for ages anyways

**Pull request merged**